react/features/local-recording/recording/WavAdapter.js

```diff
@@ -0,0 +1,284 @@
+import { RecordingAdapter } from './RecordingAdapter';
+import { downloadBlob, timestampString } from './Utils';
+
+const logger = require('jitsi-meet-logger').getLogger(__filename);
+
+const WAV_BITS_PER_SAMPLE = 16;
+const WAV_SAMPLE_RATE = 44100;
+
+/**
+ * Recording adapter for raw WAVE format.
+ */
+export class WavAdapter extends RecordingAdapter {
+
+    _audioContext = null;
+    _audioProcessingNode = null;
+    _audioSource = null;
+
+    _wavLength = 0;
+    _wavBuffers = [];
+    _isInitialized = false;
+
+    /**
+     * Constructor.
+     *
+     */
+    constructor() {
+        super();
+
+        this._saveWavPCM = this._saveWavPCM.bind(this);
+    }
+
+    /**
+     * Implements {@link RecordingAdapter#ensureInitialized()}.
+     *
+     * @inheritdoc
+     */
+    ensureInitialized() {
+        if (this._isInitialized) {
+            return Promise.resolve();
+        }
+
+        const p = new Promise((resolve, reject) => {
+            navigator.getUserMedia(
+
+                // constraints - only audio needed for this app
+                {
+                    audioBitsPerSecond: WAV_SAMPLE_RATE * WAV_BITS_PER_SAMPLE,
```

Comment:
Does `getUserMedia` support an `audioBitsPerSecond` constraint? I only see it defined for the `MediaRecorder` API.